File: vertx-examples/src/main/ruby/mem_usage/mem.rb
require "vertx"


def deploy_it(count)
  Vertx::deploy_verticle("child#{count}.rb") do |deploy_id|
    puts "deployed #{count}"
    undeploy_it(deploy_id, count)
  end
end

def undeploy_it(deploy_id, count)
  #puts "undeploying #{deploy_id}"
  Vertx::undeploy_verticle(deploy_id) do
    # puts "undeployed"
    count = count + 1
    if count < 1000000
      deploy_it(count)
    else
      puts "done!"
    end
  end
end


# Vertx::deploy_verticle('child.rb')
deploy_it(0)

File: vertx-examples/src/main/javascript/mem_usage/mem.js
load('vertx.js');

function deploy_it(count) {
  vertx.deployVerticle('child' + count + '.js', null, 1, function(deploy_id) {
    console.log("deployed " + count);
    undeploy_it(deploy_id, count);
  });
}

function undeploy_it(deploy_id, count) {
  console.log("undeploying " + deploy_id);
  vertx.undeployVerticle(deploy_id, function() {
    console.log("undeployed");
    count++;
    if (count < 100000) {
      deploy_it(count);
    } else {
      console.log("done!");
    }
  });
}


vertx.deployVerticle('child.js');
deploy_it(0) ;